### PR TITLE
feat: disallow targetIdentityId option in all S3 write APIs and category API

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

* Disallow `targetIdentityId` option from all write APIs in storage. Previously when use `uploadData` or `remove` API inputs uses `type RemoveOptions = Omit<StorageOptions, 'targetIdentityId'>` to patch the extra `targetIdentityId`. However, it breaks in TS 3.8 because in this case `RemoveOptions` is no longer extended from `StorageOptions`.
* Since `identityId` is a AWS concept, removing it from all the category level interfaces

#### Description of how you validated changes
* unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/providers/s3/types/options.ts [hunk 1]
```diff
@@ -1,89 +1,121 @@
...
-	StorageOptions,
 	StorageListAllOptions,
 	StorageListPaginateOptions,
 } from '../../../types/options';
 
-/**
- * Input options type for S3 Storage operations.
- */
-export type Options = StorageOptions & {
+type S3SharedOptions = {
```
Comment (ashwinkumar6) [inline packages/storage/src/providers/s3/types/options.ts:14]: nit
I'm wondering if we should remove the `S3` prefix from `S3SharedOptions`, `S3ReadOptions`, `S3WriteOptions`

**Review by ashwinkumar6 — APPROVED** [review]
LGTM, just one nit

File: packages/storage/src/providers/s3/types/options.ts (hunk 1 above)
Comment (AllanZhengYP) [reply packages/storage/src/providers/s3/types/options.ts:14]: Yeah I don't feel strongly about it either. will change it.

**Review by ashwinkumar6 — APPROVED** [review]

**Review by stocaaro — DISMISSED** [review]

**Review by ashwinkumar6 — APPROVED** [review]

**Review by stocaaro — APPROVED** [review]

**Comment by johnf** [comment]
Would you consider exposing this again?

My use case is that I use the username instead of the identity to store the files.

My rationale here is that I was bitten in the past when I had to recreate my cognito auth, in a new Cogntio if you backu and restore you can set the username but not the identity.

**Pull request merged**